### ISSUE TITLE
Ask for the type of this algorithm

### ISSUE THREAD

**Issue by Cloud65000** [wanghao9610/OV-DINO#33]
Hi! I successfully ran the whole process of inference and went deep to read almost all the code of the algorithm. Since I first ran the Groundingdino and then the OVDINO, I realized that OVDINO is more likely to be a CLIP-based algorithm. I'm not sure if I can call it "CLIP_Based" but there's no deep fusion between two modalities but several constructive dot products between two kinds of embedding. I just want to make sure whether my opinion is correct or not.

**Comment by wanghao9610** [comment]
@Cloud65000 Hello, OV-DINO is not based on the CLIP pre-trained model, while we pre-train the model on large-scale datasets and explore the region-level vision-language alignment (while CLIP is image-level alignment). OV-DINO shares similar vision-language contrastive learning (classification loss can be regarded as a special case of contrastive loss), and we are motivated by CLIP a lot, such as region-text alignment, unified prompt.

**Comment by Cloud65000** [comment]
I see. So there's no fusion between the two modalities right?

**Comment by Cloud65000** [comment]
I just see the cross-attention done between the label((appear like 0,1,2,3,not the specific text description of the categories) embeddings and the image features

> I see. So there's no fusion between the two modalities right?

**Comment by wanghao9610** [comment]
> I see. So there's no fusion between the two modalities right?

Language-aware selective fusion is our main contribution of our paper, which contains the cross-modality fusion in the decoder, you will find the details in the [paper](https://arxiv.org/abs/2407.07844).

**Comment by Cloud65000** [comment]
Sorry, I found the code.
![image](https://github.com/user-attachments/assets/9c0b02a8-05af-445a-9d43-b34ba8ab2ed4)
There are several differences between the LASF of OVDINO and "the  language-guided query selection" of GDINO. I  was kind of confusing the two. I will read the code carefully again.

**Comment by Cloud65000** [comment]
![image](https://github.com/user-attachments/assets/91c5da4c-e0c8-4530-9796-65f7c941ec9d)
I recalled the code I'd read and viewed the picture in the paper, there was no obvious description of how to use text embeddings coming out from the text backbone except to calculate the dot product between the two modalities and do the top 900 filtering for the image features coming from the encoder.

**Comment by wanghao9610** [comment]
> ![image](https://private-user-images.githubusercontent.com/147381573/362639154-91c5da4c-e0c8-4530-9796-65f7c941ec9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5Mjg2NTgsIm5iZiI6MTcyNDkyODM1OCwicGF0aCI6Ii8xNDczODE1NzMvMzYyNjM5MTU0LTkxYzVkYTRjLWUwYzgtNDUzMC05Nzk2LTY1ZjdjOTQxZWM5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOVQxMDQ1NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMGI0Nzc0ZDUyMDAyMjg5N2UxOWExNWJmMGFmOGRjMzk1ZDFjYWQ1ZDU0ZjU1NzEzMGZjYjk4NmVmMzdjMjNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5QstCcOfTGYq-Q6Hft7m0fbJymL2lfejY-GvIp9qWWo) I recalled the code I'd read and viewed the picture in the paper, there was no obvious description of how to use text embeddings coming out from the text backbone except to calculate the dot product between the two modalities and do the top 900 filtering for the image features coming from the encoder.

You should read the [dino_transformer.py](https://github.com/wanghao9610/OV-DINO/blob/main/ovdino/projects/ovdino/modeling/dino_transformer.py) from L832 to the end, you will find the implementation of LASF (LAQS followed LAQF). As for the paper, you will find the details on Sec. 3.3 .

**Comment by Cloud65000** [comment]
I read the code of [dino_transformer.py](https://github.com/wanghao9610/OV-DINO/blob/main/ovdino/projects/ovdino/modeling/dino_transformer.py) again. Here's my understanding: the text features coming from the text backbone don't explicitly participate in the fusion between two modalities. They just perform as the teacher who guides the model to find the text-related image features(like reference_points and target_unact in the line 840 and  848 of [dino_transformer.py](https://github.com/wanghao9610/OV-DINO/blob/main/ovdino/projects/ovdino/modeling/dino_transformer.py) . Then the GatedMultiheadAttention function will fuse the image features initially coming from the encoder and the image features selected by top k method and the latter one implicitly carries the information of the text features. 
Have I understood correctly?

**Comment by wanghao9610** [comment]
> I read the code of [dino_transformer.py](https://github.com/wanghao9610/OV-DINO/blob/main/ovdino/projects/ovdino/modeling/dino_transformer.py) again. Here's my understanding: the text features coming from the text backbone don't explicitly participate in the fusion between two modalities. They just perform as the teacher who guides the model to find the text-related image features(like reference_points and target_unact in the line 840 and 848 of [dino_transformer.py](https://github.com/wanghao9610/OV-DINO/blob/main/ovdino/projects/ovdino/modeling/dino_transformer.py) . Then the GatedMultiheadAttention function will fuse the image features initially coming from the encoder and the image features selected by top k method and the latter one implicitly carries the information of the text features. Have I understood correctly?

Your understanding is right, but there exists little misunderstanding: the text embedding is not as a teacher, while it provides language-aware information for object embedding selection and later fusion.
